modules/hal_nordic/nrf_802154/CMakeLists.txt

```diff
@@ -15,10 +15,10 @@ if (CONFIG_NRF_802154_RADIO_DRIVER)
       ${CMAKE_CURRENT_SOURCE_DIR}/sl_opensource/platform/nrf_802154_temperature_zephyr.c
   )
 
-  if (CONFIG_SOC_SERIES_NRF53X AND NOT CONFIG_NRF_802154_SL_OPENSOURCE)
+  if (NOT CONFIG_NRF_802154_SL_OPENSOURCE)
     target_sources(nrf-802154-platform
       PRIVATE
-        ${CMAKE_CURRENT_SOURCE_DIR}/sl_opensource/platform/nrf_802154_lp_timer_zephyr.c
+        ${CMAKE_CURRENT_SOURCE_DIR}/sl_opensource/platform/nrf_802154_platform_sl_lptimer_zephyr.c
```

Comment:
I think the platform file could be moved directly to sdk-nrf. This would reduce the path that modifications to this file have to take.

Reply:
@ankuns sorry for the delayed response.

In my opinion it's much more convenient to have all platform files for a given platform in a single place. As you mentioned, even now [this directory](https://github.com/zephyrproject-rtos/zephyr/tree/main/modules/hal_nordic/nrf_802154/sl_opensource/platform) contains both files that are needed by opensource variant of the 802.15.4 driver as well as those that are only needed by the full-featured variant. I'd prefer to have a consistent pattern of storing these platform files.

I think it's fine to split these platform files into those needed by all variants (stored somewhere upstream) and those needed by the full-featured variant only (stored somewhere downstream), but I'd like to avoid moving just one file to a different repo, because such inconsistency is difficult to follow in the long run.

What do you think?

Reply:
Agree, we need consistent approach.

Here I'm OK to merge as it is now.

Reply:
@e-rk has already moved the file to sdk-nrf. Are we going to keep it this way to merge the PRs quicker or roll back to the original suggestion (all platform files stored upstream)?

Reply:
At current state I think we should get it merged quick.

Reply:
OK, I'm approving the PR then. Let's get back to the discussion about platform files after this PR is merged